docs/design/clarifying-object-strucutre.md

```diff
@@ -0,0 +1,111 @@
+# Clarifying object structure
+
+Today we have resources that look like:
+```yaml
+apiVersion: microsoft.storage.infra.azure.com/v1alpha1api20190401
+kind: StorageAccount
+metadata:
+  name: samplekubestorage
+  namespace: default
+spec:
+  azureName: mykubestorage
+  location: westcentralus
+  kind: BlobStorage
+  sku:
+    name: Standard_LRS
+  owner:
+    name: k8sinfra-sample-rg
+  properties:
+    accessTier: Hot
+  tags:
+    tag1: tag1
+    tag2: tag2
+```
+
+There is ongoing work to flatten the stuff in `spec.properties` up a level into `spec`:
+```yaml
+apiVersion: microsoft.storage.infra.azure.com/v1alpha1api20190401
+kind: StorageAccount
+metadata:
+  name: samplekubestorage
+  namespace: default
+spec:
+  azureName: mykubestorage
+  location: westcentralus
+  kind: BlobStorage
+  sku:
+    name: Standard_LRS
+  owner:
+    name: k8sinfra-sample-rg
+  accessTier: Hot
+  tags:
+    tag1: tag1
+    tag2: tag2
+```
+
+## The problem
+There's no good way with this object structure to differentiate stuff that is for Azure directly, versus stuff that is
+for the operator. `owner` _almost_ falls into this category already, but there are other likely upcoming properties that
+definitely fall into this category: 
+ - `SecretConfiguration`: details about where/how we should store secrets created by this entity.
+ - `Credentials`: per object credentials used to support CAPZ-like scenarios.
+
+The problem also manifests in `Status` where ideally we would distinguish between properties from Azure directly (the
+result of a GET on the resource) and properties that we are presenting. For example, we may want to have a status field
+for `deploymentId` documenting the deployment ID used to create the resource. If we do that there's no easy way for the
+customer to understand that field is provided by ASO and not by Storage.
+
+## Proposal
+We introduce an additional level of hierarchy specifically to clarify what is coming or going directly from or to Azure.
+This is similar to what Crossplane does (see their 
+[MySQLServer](https://github.com/crossplane/provider-azure/blob/master/examples/database/mysqlserver.yaml)).
+
+On the `spec` we could call this something like `forAzure` or `azureSpec`, and on the `status` `fromAzure` or `azureStatus`.
```

Comment:
I think this has come up in discussion elsewhere, but worth capturing here - this pushes all the stuff that customers are interested in _down_ a level. Maybe we should instead push ASO stuff down into a container?

Reply:
Yes, I will update the spec to reflect this